### PR TITLE
Add proposal for ContainerPriority

### PR DESCRIPTION
**Pull request by FillZpp**
Signed-off-by: FillZpp <FillZpp.pub@gmail.com>

### Ⅰ. Describe what this PR does
Add proposal for ContainerLaunchPriority, which controls the sequence of containers start in a Pod.

Read link: https://github.com/FillZpp/kruise/blob/add-proposal-for-containers-starting-priority/docs/proposals/20210809-containerpriority.md

### CODE REVIEW

File: docs/proposals/20210809-containerpriority.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+      value: "1"
+    - name: KRUISE_CONTAINER_BARRIER
+      valueFrom:
+        configMapKeyRef:
+          name: {pod-name}-barrier
+          key: "c_1"   # its KRUISE_CONTAINER_PRIORITY is 1
+    # ...
+```
+
+*Note that the key of containers will be `c_{priority}` and the key of init containers will be `i_{priority}`.*
```
Comment (furykerry) [inline docs/proposals/20210809-containerpriority.md:107]: the start sequence of init container is already deterministic, i don't think it is necessary to mess with the init containers

File: docs/proposals/20210809-containerpriority.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    # ...
+    env:
+    - name: KRUISE_CONTAINER_BARRIER
+      valueFrom:
+        configMapKeyRef:
+          name: {pod-name}-barrier
+          key: "c_0"   # no KRUISE_CONTAINER_PRIORITY, so defaults to 0
+  - name: sidecar
+    env:
+    - name: KRUISE_CONTAINER_PRIORITY
```
Comment (furykerry) [inline docs/proposals/20210809-containerpriority.md:97]: is it necessary to include integer value priority in the environment? is a simple boolean value such as KRUISE_CONTAINER_STRONG_ORDER suffice ? the containers list of the pod spec is already ordered, and kubelet will start the containers in order, if the orders in containers list is conflicts with the one specified by environment variable , the orders can be hard to reason about.

**Review by furykerry — CHANGES_REQUESTED** [review]

File: docs/proposals/20210809-containerpriority.md (hunk 1 above)
Comment (FillZpp) [reply docs/proposals/20210809-containerpriority.md]: Removed, don't need to deal with init containers.

**Pull request merged**